src/github/tests/test_utils.rs

```diff
@@ -89,22 +103,82 @@ impl DataModel {
             }
         }
 
+        let mut repos = HashMap::default();
+        let mut repo_members: HashMap<String, (Vec<RepoTeam>, Vec<RepoUser>)> = HashMap::default();
+
+        for repo in &self.repos {
+            repos.insert(
+                repo.name.clone(),
+                Repo {
+                    repo_id: repos.len() as u64,
+                    node_id: repos.len().to_string(),
+                    name: repo.name.clone(),
+                    org: DEFAULT_ORG.to_string(),
+                    description: Some(repo.description.clone()),
+                    homepage: repo.homepage.clone(),
+                    archived: false,
+                    allow_auto_merge: None,
+                },
+            );
+            let teams = repo
+                .teams
+                .clone()
+                .into_iter()
+                .map(|t| api::RepoTeam {
+                    name: t.name,
+                    permission: match t.permission {
+                        RepoPermission::Write => api::RepoPermission::Write,
+                        RepoPermission::Admin => api::RepoPermission::Admin,
+                        RepoPermission::Maintain => api::RepoPermission::Maintain,
+                        RepoPermission::Triage => api::RepoPermission::Triage,
+                    },
+                })
+                .collect();
+            let members = repo
+                .members
+                .clone()
+                .into_iter()
+                .map(|m| api::RepoUser {
+                    name: m.name,
+                    permission: match m.permission {
+                        RepoPermission::Write => api::RepoPermission::Write,
+                        RepoPermission::Admin => api::RepoPermission::Admin,
+                        RepoPermission::Maintain => api::RepoPermission::Maintain,
+                        RepoPermission::Triage => api::RepoPermission::Triage,
```

Comment:
this code block is duplicated. can we implement the From trait to extract it?

Reply:
I reused a function from the GH API instead; we want to reuse as much code as possible anyway, to avoid introducing test-specific behavior.